token/services/interop/exchange/script.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package exchange
+
+import (
+	"crypto"
+	"encoding/json"
+	"time"
+
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/view"
+	"github.com/hyperledger-labs/fabric-token-sdk/token"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/core/identity"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/interop/encoding"
+	token3 "github.com/hyperledger-labs/fabric-token-sdk/token/token"
+)
+
+// HashInfo contains the information regarding the hashing
+type HashInfo struct {
+	Hash         []byte
+	HashFunc     crypto.Hash
+	HashEncoding encoding.Encoding
+}
+
+// Script contains the details of an exchange
+type Script struct {
+	Sender    view.Identity
+	Recipient view.Identity
+	Deadline  time.Time
+	HashInfo  HashInfo
+}
+
+// ScriptOwnership implements the Ownership interface for scripts
+type ScriptOwnership struct{}
+
+// AmIAnAuditor returns false for script ownership
+func (s *ScriptOwnership) AmIAnAuditor(tms *token.ManagementService) bool {
+	return false
+}
+
+// IsMine returns true it there exists an owner wallet for the token's owner
```

Comment:
This is not very clear. According to the code, IsMine checks if one is either a sender or recipient in exchange script.